test/e2e/live_iso_test.go

```diff
@@ -30,25 +30,43 @@ func liveIsoTest() {
 	It("Should update the bmh with live iso", func() {
 		bootstrapClient := bootstrapClusterProxy.GetClient()
 
+		By("Waiting for bmhs to be available")
+		Eventually(
+			func(g Gomega) {
+				bmhs, err := getAllBmhs(ctx, bootstrapClient, namespace, specName)
+				Expect(err).NotTo(HaveOccurred(), "Error getting BMHs")
+				filtered := filterBmhsByProvisioningState(bmhs, bmov1alpha1.StateAvailable)
+				g.Expect(len(filtered)).To(Equal(len(bmhs)))
+				Logf("one or more BMHs are not available yet")
```

Comment:
is this needed? this is misleading, we are expecting all bmhs to be in available just before this,

Reply:
this is just to make sure previous test's deprovisioning is not in process. We thought this might be the issue but now we have checked that the issue was with BMHs Online status, we can remove it, but I think precheck makes the test more robust. Let me know what do you think, should we keep it or remove it?

Reply:
Fine by me to leave it